### DISCUSSION TITLE
Flutter - Web Integration tests - "chromedriver installation failed"

### DISCUSSION THREAD

**Discussion by ArmandF-Eclipsed** [codemagic-ci-cd/codemagic-docs#2551] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1. Create a new workflow
2. Set it to only run tests
3. Enable integration tests for Chrome (Flutter drive)
4. Specify the driver file location and target test, in the parameter field.
5. Run workflow
6. Test fails with message "chromedriver installation failed"

### Expected results

Expected the Integration test to run just like it does on the local system or Github actions.

### Actual results

The testing step fails after a second with the message: 

```
== QA ==


QA failed :|
chromedriver installation failed
```

### Build id (optional)

664f7ce9adae5c26f29acda4

**Comment by himesh-cm** [comment]
Hey @ArmandF-Eclipsed ,
There is a failure when attempting to retrieve the chrome zip file from the following path: `****/chrome-for-testing/123.0.6312.122/mac-arm64/chromedriver-mac-arm64.zip`. I am actively working on reproducing the issue and will provide updates accordingly. In the meantime, could you please provide more information about the unit tests you are attempting to run? This additional information will assist in further investigating the issue.

Also, Please note that you should contact our team through the chat widget in the UI for a quicker response, as your team is on a paid plan.

**Reply by ArmandF-Eclipsed** [reply]
Hi @himesh-cm, 

Thanks for coming back to me regarding my issue.

I'm trying to run, a standard flutter integration test. The idea is to run it on the web, using the flutter drive command. The integration test driver config is also added to the project root and all of the tests can run on local and on GitHub actions. I even went as far as SSH-ing onto the machine, starting the chrome driver, then running the flutter drive command in the folder, and it seems to run correctly.

Correct! Currently, we are just looking into where we can run our integration tests, and as such it's not a priority problem for us at this moment, that's why we decided to take it to the issue/Q&A forum.

**Comment by icarusdust** [comment, marked as answer]
Hi @ArmandF-Eclipsed, upgrade `chromedriver` and chrome browser versions by running the following scripts before executing the test command:

```
brew upgrade --cask chromedriver
brew install --cask google-chrome
```
Refer to the screenshot on where to add the scripts. Could be Pre-test or Post-clone:
<img width="966" alt="Screenshot 2024-06-03 at 11 39 22" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/43378371/cb51df87-129f-41a8-83b4-cc5bdfb4701e">

They will be upgraded from our end as well, we added it to the backlog. For now, these scripts will help you continue your workflows.

**Reply by ArmandF-Eclipsed** [reply]
Thanks for the help. Adding the homebrew update before the test runs works!
